File: 7kyu/testing_1-2-3.ts
export function number(array: string[]): string[] {
  return array.map((str, i) => `${i + 1}: ${str}`);
}

/*
Parameters: an array of strings
Return: an array of strings
Examples:
  number([]) => [];
  number(["a", "b", "c"]) => ["1: a", "2: b", "3: c"]; 
  number(["", "", "", "", ""]) => ["1: ", "2: ", "3: ", "4: ", "5: "]; 
Pseudocode:
  return an array, mapped over from the given array with the string prepended with index + 1 and a colon and space
*/

File: 7kyu/testing_1-2-3.py
# My solution

def number(lines):
    return [f"{i + 1}: {line}" for i, line in enumerate(lines)]
    
'''
Parameters: a list of strings
Return: a list of strings
Examples:
    number([]) -> []
    number(["a", "b", "c"]) -> ["1: a", "2: b", "3: c"]
Pseudocode:
    Iterate over the given list
        insert the index + 1, a colon, and a space
    return the new list
'''

# Top solution

def number(lines):
    return [f"{counter}: {line}" for counter, line in enumerate(lines, start=1)]